lapack/testlapack/dtrexc.go

```diff
@@ -205,16 +199,126 @@ func testDtrexc(t *testing.T, impl Dtrexcer, compq lapack.UpdateSchurComp, tmat
 				continue
 			}
 			if q.Data[i*q.Stride+j]-qCopy.Data[i*qCopy.Stride+j] != 0 {
-				t.Errorf("%v: unexpected modification of Q[%v,%v]", prefix, i, j)
+				t.Errorf("%v: unexpected modification of Q[%v,%v]", name, i, j)
 			}
 		}
 	}
-	// Check that Qᵀ TOrig Q == T.
-	tq := eye(n, n)
-	blas64.Gemm(blas.NoTrans, blas.NoTrans, 1, tmatCopy, q, 0, tq)
-	qtq := eye(n, n)
-	blas64.Gemm(blas.Trans, blas.NoTrans, 1, q, tq, 0, qtq)
-	if !equalApproxGeneral(qtq, tmat, tol) {
-		t.Errorf("%v: Qᵀ (initial T) Q and (final T) are not equal", prefix)
+	// Check that Qᵀ * TOrig * Q == T
+	qt := zeros(n, n, n)
+	blas64.Gemm(blas.Trans, blas.NoTrans, 1, q, tmatCopy, 0, qt)
+	qtq := cloneGeneral(tmat1)
+	blas64.Gemm(blas.NoTrans, blas.NoTrans, -1, qt, q, 1, qtq)
+	resid = dlange(lapack.MaxColumnSum, n, n, qtq.Data, qtq.Stride)
+	if resid > float64(n)*tol {
+		t.Errorf("%v: mismatch between Qᵀ*(initial T)*Q and (final T); resid=%v, want<=%v",
+			name, resid, float64(n)*tol)
+	}
+}
+
+func residualOrthogonal(q blas64.General, rowwise bool) float64 {
```

Comment:
Yes, this is yet another way in `testlapack` to check orthogonality of matrices. I'll remove the others and change the test to use this one.

Reply:
ok